iocore/cache/P_CacheVol.h

```diff
@@ -276,11 +276,18 @@ struct CacheVol {
 
 // Note : hdr() needs to be 8 byte aligned.
 struct Doc {
-  uint32_t magic;        // DOC_MAGIC
-  uint32_t len;          // length of this fragment (including hlen & sizeof(Doc), unrounded)
-  uint64_t total_len;    // total length of document
-  CryptoHash first_key;  ///< first key in object.
-  CryptoHash key;        ///< Key for this doc.
+  uint32_t magic;     // DOC_MAGIC
+  uint32_t len;       // length of this fragment (including hlen & sizeof(Doc), unrounded)
+  uint64_t total_len; // total length of document
+#ifndef TS_ENABLE_FIPS
+  CryptoHash first_key; ///< first key in object.
+  CryptoHash key;       ///< Key for this doc.
+#else
+  // For FIPS CryptoHash is 256 bits vs. 128, and the 'first_key' must be checked first, so
+  // ensure that the new 'first_key' overlaps the old 'first_key' and that the rest of the data layout
+  // is the same by putting 'key' at the ned.
```

Comment:
'ned'?

Reply:
I'm unclear on why the rest of the `Doc` layout needs to be consistent. If the cache is effectively reset then it would seem to be irrelevant.

Reply:
The idea is that we would just not be able to see any of the old objects.  While it might work to not have the layout consistent, that would make a strong dependency that the inconsistent layout members where never accessed before the first_key is checked.  While I believe that is the case, this is safer in that it doesn't have the dependency.